src/js/components/Carousel.js

```diff
@@ -58,6 +70,33 @@ export default class Carousel extends Component {
     scrollParents.forEach(function (scrollParent) {
       scrollParent.removeEventListener('scroll', this._handleScroll);
     }.bind(this));
+
+    this._unmountHammer();
+  }
+
+  _unmountHammer () {
+    if (this.hammer) {
```

Comment:
I see. it seems that when invoking functions from inside react lifecycle (componentDidMount) then you dont need the binding context... anyways thanks for the explanation.